internal/controller/scheduler/scheduler.go

```diff
@@ -36,9 +36,13 @@ func (scheduler *Scheduler) Run() {
 		case <-scheduler.schedulingRequested:
 		case <-time.After(schedulerInterval):
 		}
+
 		if err := scheduler.schedulingLoopIteration(); err != nil {
 			scheduler.logger.Errorf("Failed to schedule VMs: %v", err)
 		}
+		if err := scheduler.healthCheckingLoopIteration(); err != nil {
```

Comment:
I think it makes sense to do health checking before scheduling and not after? In most cases it will be a no-op: just load all the VMs and don't update any of them.

Reply:
Moved in ee5f192.